src/components/pager/pager-navigation.component.js

```diff
@@ -1,217 +1,132 @@
+      setCurrentPage(1);
+      onPagination(1, pageSize, 'input');
+      return '1';
     }
 
-    return (
-      <PagerButtonWrapperStyles disabled={ disabled() } next={ step === 1 }>
-        <Icon
-          onClick={ () => {
-            if (!disabled()) { navArrowChange(step); }
-          } }
-          { ...arrowProps }
-        />
-      </PagerButtonWrapperStyles>
-    );
-  }
-
-  function navLink(type) {
-    const currentPage = Number(props.currentPage);
-    const navLinkConfig = {
-      first: {
-        text: I18n.t('pager.first', { defaultValue: 'First' }),
-        destination: '1'
-      },
-      last: {
-        text: I18n.t('pager.last', { defaultValue: 'Last' }),
-        destination: String(maxPages())
-      },
-      next: {
-        text: I18n.t('pager.next', { defaultValue: 'Next' }),
-        destination: String(currentPage + 1)
-      },
-      back: {
-        text: I18n.t('pager.previous', { defaultValue: 'Previous' }),
-        destination: String(currentPage - 1)
-      }
-    };
-
-    const { destination, text } = navLinkConfig[type];
-
-    const clickHandler = () => {
-      onPagination(destination, props.pageSize, type);
-    };
-
-    function disabled() {
-      if (maxPages() === 1) {
-        return true;
-      }
-      if (currentPage === 1) {
-        return type === 'back' || type === 'first';
-      }
-      if (currentPage === maxPages()) {
-        return type === 'next' || type === 'last';
-      }
-      return false;
+    if (Number(ev.target.value) > pageCount) {
+      setCurrentPage(pageCount);
+      onPagination(pageCount, pageSize, 'input');
+      return pageCount;
     }
 
-    return (
-      <PagerLinkStyles
-        disabled={ disabled() }
-        onClick={ !disabled() ? clickHandler : undefined }
-      >
-        { text }
-      </PagerLinkStyles>
-    );
+    setCurrentPage(Number(ev.target.value));
+    onPagination(Number(ev.target.value), pageSize, 'input');
+    return ev.target.value;
   }
 
-  function handlePageInputChange(ev) {
-    setCurrentPage(ev.target.value);
-  }
+  const handleCurrentPageChange = (e) => {
+    setCurrentPage(e.target.value);
+  };
 
   function currentPageInput() {
     const currentPageInputProps = {
-      value: props.currentPage,
+      value: currentPage.toString(),
       className: 'carbon-pager__current-page',
       'data-element': 'current-page',
-      onChange: handlePageInputChange,
-      onBlur: updatePageFromInput,
+      onChange: handleCurrentPageChange,
+      onBlur: handlePageInputChange,
       onKeyUp: (ev) => {
         if (!Events.isEnterKey(ev)) {
           return false;
         }
-        return updatePageFromInput(ev);
+        return handlePageInputChange(ev);
       }
     };
 
-    if (isClassic(theme)) {
-      return (
-        <label>
-          <NumberComponent { ...currentPageInputProps } />
-        </label>
-      );
-    }
     return (
       <label>
```

Comment:
its inconclusive of how it can be fixed, I did the recommended and added htmlFor along with matching id to input. But I had to put in ignore of eslint to get rid of pesky warning.